Fix unit tests for Data Grid 7.3 image

osnipso opened the pull request:
TODO: fix the s2i builds tests, squash commits before merging

Thanks for submitting your Pull Request!

Please make sure your PR meets the following requirements:

- [X] Pull Request contains description of the issue
- [X] Pull Request does not include fixes for issues other than the main ticket
- [X] Attached commits represent units of work and are properly formatted
- [X] You have read and agreed to the Developer Certificate of Origin (DCO) (see `CONTRIBUTING.md`)
- [X] Every commit contains `Signed-off-by: Your Name <yourname@example.com>` - use `git commit -s`

ryanemerson commented on image.yaml:
```diff
@@ -1,232 +1,234 @@
-                  url: https://github.com/jboss-openshift/cct_module.git
-                  ref: master
-          - name: datagrid.72.modules
-            path: modules/datagrid/72
```
As we no longer use any of the modules in `modules/datagrid/72`, we should delete them. Also, I think we should remove the version dir completely. We're using branches to differentiate between versions, therefore by having `72` etc, it means that every time we move to the next DG version we have to mess up our git diff as files are moved. If we remove the version, this is no longer an issue.

Also, a general comment about the module structure. I like the idea of having the `common` directory to show what has come from `cct_modules`, however I don't think this should be a subdir of datagrid. Instead how about the following structure:

```
modules/
   - datagrid/
      - configuration
      - distribution
      - launch
   - common/
      - datagrid-7-openshift
     ....
```

Long term as we simplify the modules inherited from `cct_modules` and remove the scripts/configuration options that are redundant to DG we can reduce the size of the common dir.

osnipso replied:
I agree with both points, will do.

ryanemerson commented on modules/datagrid/73/configuration/added/clustered-openshift.xml:
```diff
@@ -0,0 +1,320 @@
+            </datasources>
+        </subsystem>
+<!-- ##INFINISPAN_ENDPOINT## -->
+        <subsystem xmlns="urn:infinispan:server:core:8.5" default-cache-container="clustered">
```
We should update `clustered-openshift.xml` to utilise all the latest schema versions supported by 7.3. Which is 9.4 for the `core`, `endpoint` and `jgroups` subsystems. We should also update the versions of the other other subsystems. See this gist for a config which contains the latest versions of subsystems: https://gist.github.com/ryanemerson/a0fb20ad5a39aefb300ca216b38142f1.

osnipso replied:
Sure, I thought so. I'll try to find the `.xsd` files, like you mentioned, so to know about this for the next versions. Will do.

ryanemerson commented on modules/datagrid/73/launch/added/launch/infinispan-config.sh:
```diff
@@ -0,0 +1,745 @@
+  fi
+
+  subsystem="\
+        <subsystem xmlns=\"urn:infinispan:server:endpoint:8.1\">$hotrod $memcached $rest\
```
This will have to be updated to reference 9.4 once the `clustered-openshift.xml` has been updated to use the latest version.

osnipso replied:
Sure, makes sense. Will do.

ryanemerson commented:
@osnipso A more general point, can we remove all of the `README.adoc` files that have been committed. AFAICT these don't add anything and just adds noise to the PR and our workspace.

Regarding the modules paths, I don't think we need to update all references as not all of the modules have been relocated to `${JBOSS_HOME}/modules/system/add-ons`, only the jars specific to Infinispan. For example the logmanager changes are probably not required, because  when I do `find modules -path "**/logmanager/**/*.jar"` on the DR4 server structure, the following is returned:

```
modules/system/layers/base/org/jboss/log4j/logmanager/main/log4j-jboss-logmanager-1.1.6.Final-redhat-00001.jar
modules/system/layers/base/org/jboss/logmanager/commons/logging/main/commons-logging-jboss-logmanager-1.0.3.Final.jar
modules/system/layers/base/org/jboss/logmanager/main/jboss-logmanager-2.1.4.Final-redhat-00001.jar
```

osnipso commented:
> @osnipso A more general point, can we remove all of the `README.adoc` files that have been committed. AFAICT these don't add anything and just adds noise to the PR and our workspace. 

Nice, I had this question when working on migrating the modules, as in `https://pastebin.com/jLbskGYz`, line 80. Will get rid of them.

> Regarding the modules paths, I don't think we need to update all references as not all of the modules have been relocated to `${JBOSS_HOME}/modules/system/add-ons`, only the jars specific to Infinispan. For example the logmanager changes are probably not required, because when I do `find modules -path "**/logmanager/**/*.jar"` on the DR4 server structure, the following is returned:
> 
> ```
> modules/system/layers/base/org/jboss/log4j/logmanager/main/log4j-jboss-logmanager-1.1.6.Final-redhat-00001.jar
> modules/system/layers/base/org/jboss/logmanager/commons/logging/main/commons-logging-jboss-logmanager-1.0.3.Final.jar
> modules/system/layers/base/org/jboss/logmanager/main/jboss-logmanager-2.1.4.Final-redhat-00001.jar
> ```

As per our discussion in Zulip, I tried to do so in the beginning, but stumble upon the way the `openshift-layer` module works when linking to the jar files. Will give it another try, if it's too complicated I'll leave it as it is (provided it does work and load the logmanager jars) and we can refactor it as we go.

Also, I'll wait until PRs #134 and #135 are merged, rebase all this work over #135 and get rid of the commits made redundant by it.

galderz commented:
Is this WIP? Or can it be reviewed?

ryanemerson reviewed (changes requested):
@osnipso The `CommandLine` class in our testsuite needs to be updated to utilise `./bin/ispn-cli.sh` instead of `./bin/cli.sh` as the latter no longer exists.

Also, as this is an initial import, can you make all of the `*.yaml` files use the same indentation. Some of the common modules still use the old format e.g. `dynamic-resources/module.yaml`

@tristantarrant For backwards compatibility do we need to provide `cli.sh` as an alias in 7.3?

ryanemerson commented:
@osnipso FYI, it has been established in our meeting that the `cli` command should be `cli.sh`, not `ispn-cli.sh` in the server distribution. However, as that is not currently the case in DR4, and it's trivial to update `CommandLine.java`, we should update the tests and revert to `cli.sh` in future releases.

osnipso commented:
@ryanemerson Changed the CommandLine class for the tests and tried to follow the guidelines for yaml indentation and be consistent, PTAL.

ryanemerson commented:
@osnipso Less failures :smiley:, but there is still an issue with 3 of the tests. I'm having a look now to see if I can see what it is quickly. 

Also, I have noticed that we're pulling in a load of eap test files we don't need. The following is the result of `find modules -path "**/tests/**"`:

```
modules/common/os-eap7-launch/tests/bats/launch-common
modules/common/os-eap7-launch/tests/bats/launch-common/launch-common.bats
modules/common/os-eap7-launch/tests/bats/datasource
modules/common/os-eap7-launch/tests/bats/datasource/configure.bats
modules/common/os-eap7-launch/tests/bats/datasource/common.bash
modules/common/os-eap7-launch/tests/bats/datasource/expectations
modules/common/os-eap7-launch/tests/bats/datasource/expectations/datasources_other_nonxa.xml
modules/common/os-eap7-launch/tests/bats/datasource/expectations/datasources_sybase_pool_nonxa.xml
modules/common/os-eap7-launch/tests/bats/datasource/expectations/datasources_other_xa.xml
modules/common/os-eap7-launch/tests/bats/datasource/expectations/prefix_ibmdb2_xa.xml
modules/common/os-eap7-launch/tests/bats/datasource/expectations/prefix_postgresql_xa.xml
modules/common/os-eap7-launch/tests/bats/datasource/expectations/datasources_postgresql_nourl_xa.xml
modules/common/os-eap7-launch/tests/bats/datasource/expectations/prefix_postgresql_nonxa.xml
modules/common/os-eap7-launch/tests/bats/datasource/expectations/prefix_mysql_xa.xml
modules/common/os-eap7-launch/tests/bats/datasource/expectations/datasources_sybase_validation_nonxa.xml
modules/common/os-eap7-launch/tests/bats/datasource/expectations/prefix_oracle_xa.xml
modules/common/os-eap7-launch/tests/bats/datasource/expectations/datasources_mysql_nonxa.xml
modules/common/os-eap7-launch/tests/bats/datasource/expectations/datasources_postgresql_nonxa.xml
modules/common/os-eap7-launch/tests/bats/datasource/expectations/datasources_ibmdb2_xa.xml
modules/common/os-eap-launch/tests/bats/maven-repos
modules/common/os-eap-launch/tests/bats/maven-repos/test.out
modules/common/os-eap-launch/tests/bats/maven-repos/add-server.bats
modules/common/os-eap-launch/tests/bats/maven-repos/add-repo.bats
modules/common/os-eap-launch/tests/bats/maven-repos/configure.bats
modules/common/os-eap-launch/tests/bats/maven-repos/common.bash
modules/common/os-eap-launch/tests/bats/maven-repos/expectations
modules/common/os-eap-launch/tests/bats/maven-repos/expectations/profile_default_discovery.xml
modules/common/os-eap-launch/tests/bats/maven-repos/expectations/profile_repo2.xml
modules/common/os-eap-launch/tests/bats/maven-repos/expectations/profile_custom_discovery.xml
modules/common/os-eap-launch/tests/bats/maven-repos/expectations/profile_all_vars.xml
modules/common/os-eap-launch/tests/bats/maven-repos/expectations/server_with_username.xml
modules/common/os-eap-launch/tests/bats/maven-repos/expectations/server_repo1.xml
modules/common/os-eap-launch/tests/bats/maven-repos/expectations/profile_default.xml
modules/common/os-eap-launch/tests/bats/maven-repos/expectations/server_with_privatekey.xml
modules/common/os-eap-launch/tests/bats/maven-repos/expectations/server_repo2.xml
modules/common/os-eap-launch/tests/bats/maven-repos/expectations/server.xml
modules/common/os-eap-launch/tests/bats/maven-repos/expectations/server_with_all.xml
modules/common/os-eap-launch/tests/bats/maven-repos/expectations/profile_repo1.xml
```

galderz commented:
@osnipso What's the status of this?

Commit:
Fix unit test scenario 'jdg cache partition handling enabled'

Feature: Openshift JDG cache

Signed-off-by: Osni Oliveira <osni.oliveira@redhat.com>

Commit:
Fix unit test scenario 'Enabling Compatibility Mode in Selected Cache with Marshaller test'

Feature: OpenShift JDG Compatibility Mode tests

Signed-off-by: Osni Oliveira <osni.oliveira@redhat.com>

Commit:
Fix unit test scenarios 'Check jgroups encryption does not create invalid configuration with missing name/password'

Feature: Openshift JDG jgroups secure

Signed-off-by: Osni Oliveira <osni.oliveira@redhat.com>

Commit:
Fix unit test scenario 'readinessProbe and livenessProbe runs successfully with user security'

Feature: Openshift DataGrid tests

Signed-off-by: Osni Oliveira <osni.oliveira@redhat.com>

ryanemerson commented:
@osnipso I merged https://github.com/jboss-container-images/jboss-datagrid-7-openshift-image/pull/144, so can you just rebase this PR based upon the latest `datagrid73-dev` so that we can get the unit test fixes.

FYI we didn't have the unit tests enabled on the CI in the past because they took so long due to things being pulled from `cct_modules`. If we want to start using them on the ci we can enable them, I just need to add a couple of dependencies to the CI.

osnipso commented:
@ryanemerson have rebased and now this PR is only about the unit tests. I made it `wip` until we figure out the s2i builds tests. I don't know about enabling those on our CI, let's discuss about it. However, running them locally is a good deal of help in any development task that changes config files, so I think it's worth to fix them anyway.

ryanemerson commented:
@osnipso Yeah they're handy to have, but if we're going to embrace them then we should have them on the CI for PRs so that we don't accidentally merge changes that break things.

Commit:
CLOUD-2970 Move base layer of JDG image from jboss-openjdk to rhel7

- Bypass CE cacher to fetch the server artifact directly from Brew
- Bypass CE cacher to fetch the os-eap7-ping module artifacts directly from Brew
- Bypass CE cacher to fetch the os-eap-logging module artifacts directly from Brew/Maven

Signed-off-by: Osni Oliveira <osni.oliveira@redhat.com>

osnipso commented:
I'm closing this for now; will get back to sort the final unit tests later.
